packages/vue/jest.config.cjs

```diff
@@ -9,7 +9,7 @@ module.exports = {
   ],
   coverageThreshold: {
     global: {
-      branches: 28,
+      branches: 22,
```

Comment:
I temporarily reduced the threshold since this PR touched Auth subcomponents as part of the refactor. 

I added unit tests to the actual code changes (`useAuth` and `useAuthenticator`) with 100% coverage, so I *think* this is acceptable for the time being as these subcomponents don't even have tests in the first place. 

```
  useAuth.ts                                       |     100 |      100 |     100 |     100 |    
```

Reply:
Fine with me, it's naturally going to add more branches due to code extraction. Despite the numbers, we probably actually have more coverage since you tested the extracted code.

Reply:
Would feel better if any of the other values were increased (fn/lines/stmts)

Reply:
That's very fair. Currently I could increase `functions` by 1 and `statements`, but that probably isn't convincing.

Let me add unit tests for `Authenticator.vue`, since that is the primary consumer of the composables. I'll update you when I get it in.

Reply:
Added some tests for `authenticator.vue`. However, it's still difficult to increase the scope with current scope of refactor because of the rest of code has:
1. a lot of unused/unsupported public API ([example](https://github.com/aws-amplify/amplify-ui/blob/main/packages/vue/src/components/authenticator.vue#L147-L225)) and
1. a lot of usage relying on explicit `authState` look up ([example](https://github.com/aws-amplify/amplify-ui/blob/main/packages/vue/src/components/sign-in.vue#L120)).

Talked offline with @ioanabrooks, we will make follow up PRs to increase the coverage.